### ISSUE TITLE
Issue with using on par with poetry

### ISSUE THREAD

**Issue by vrslev** [emmett-framework/granian#72]
Hi there! When I try to add Granian to a project, it fails.

```shell
/tmp 
❯ mkdir testpoetry-granian

/tmp 
❯ cd testpoetry-granian

/tmp/testpoetry-granian 
❯ poetry init -q

/tmp/testpoetry-granian is 📦 v0.1.0 via 🐍 v3.11.0 
❯ poetry add granian==0.3.0
Creating virtualenv testpoetry-granian in /private/tmp/testpoetry-granian/.venv

Updating dependencies
Resolving dependencies... (0.1s)

Writing lock file

Package operations: 8 installs, 0 updates, 0 removals

  • Installing idna (3.4)
  • Installing sniffio (1.3.0)
  • Installing anyio (3.6.2)
  • Installing click (8.1.3)
  • Installing typer (0.7.0)
  • Installing uvloop (0.17.0)
  • Installing watchfiles (0.18.1)
  • Installing granian (0.3.0): Failed

  AssertionError

  In /Users/lev/Library/Caches/pypoetry/artifacts/60/fe/00/2931d21a92fc2dc50a454589ca558160dde1d497d46471b44c75b3c653/granian-0.3.0-cp311-cp311-macosx_11_0_arm64.whl, granian/__init__.py is not mentioned in RECORD

  at /opt/homebrew/Cellar/poetry/1.4.0/libexec/lib/python3.11/site-packages/installer/sources.py:158 in get_contents
      154│             if item.filename[-1:] == "/":  # looks like a directory
      155│                 continue
      156│ 
      157│             record = record_mapping.pop(item.filename, None)
    → 158│             assert record is not None, "In {}, {} is not mentioned in RECORD".format(
      159│                 self._zipfile.filename,
      160│                 item.filename,
      161│             )  # should not happen for valid wheels
      162│ 

/tmp/testpoetry-granian is 📦 v0.1.0 via 🐍 v3.11.2 
❯ 
```

Here are steps that I took:
```shell
cd /tmp
mkdir testpoetry-granian
cd testpoetry-granian
poetry init -q
poetry add granian==0.3.0
```

I'm using ARM Mac. Python 3.11.2, Poetry 1.4.0. Doesn't work with Granian 0.2.0 either.

**Comment by gi0baro** [comment]
It seems to me a bug in Poetry handling `maturin` packages, as:
- it runs fine with poetry 1.3.x
- it runs fine with 1.4 and `poetry config experimental.new-installer false`

I might upgrade the maturin version used for producing the wheels, but the whole thing is still quite strange to me..

**Comment by gi0baro** [comment]
This was actually a misconfig in maturin, should now be solved in granian 0.3.1

**Comment by vrslev** [comment]
@gi0baro Great! Thanks for the help ❤️
